src/main/java/com/synopsys/integration/coverity/ws/ClientAuthenticationHandlerWSS.java

```diff
@@ -7,93 +7,86 @@
-        InputStream xwssConfig = new ByteArrayInputStream(xwssConfigText.getBytes(StandardCharsets.UTF_8));
-        ClassLoader oldCCL = Thread.currentThread().getContextClassLoader();
-        try {
-            Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
-            XWSSProcessorFactory factory = XWSSProcessorFactory.newInstance();
-            xwssProcessor = factory.createProcessorForSecurityConfiguration(xwssConfig, new SecurityEnvironmentHandler());
-        } catch (XWSSecurityException se) {
-            throw new RuntimeException(se);
-        } finally {
-            Thread.currentThread().setContextClassLoader(oldCCL);
-        }
+    public ClientAuthenticationHandlerWSS(String username, String password) {
+        this.username = username;
+        this.password = password;
     }
 
-    public boolean handleFault(SOAPMessageContext mc) {
-        return true;
-    }
+    @Override
+    public boolean handleMessage(SOAPMessageContext context) {
+        Boolean outbound = (Boolean) context.get(MessageContext.MESSAGE_OUTBOUND_PROPERTY);
+        if (outbound != null && outbound) {
+            try {
+                SOAPMessage message = context.getMessage();
+                SOAPEnvelope envelope = message.getSOAPPart().getEnvelope();
+                SOAPHeader header = envelope.getHeader();
+                if (header == null) {
+                    header = envelope.addHeader();
+                }
 
-    public void close(MessageContext mc) {
-        // Do nothing
-    }
+                SOAPElement security = header.addChildElement(WSS_AUTH_LNAME, WSS_AUTH_PREFIX, WSS_AUTH_URI);
+                security.addNamespaceDeclaration(WSU_NAMESPACE_PREFIX, NAMESPACE_URI);
 
-    public Set<QName> getHeaders() {
-        QName securityHeader = new QName(WSS_AUTH_URI, WSS_AUTH_LNAME, WSS_AUTH_PREFIX);
-        HashSet<QName> headers = new HashSet<>();
-        headers.add(securityHeader);
-        return headers;
-    }
+                SOAPElement usernameToken = security.addChildElement(USERNAME_TOKEN, WSS_AUTH_PREFIX);
+                usernameToken.addAttribute(new QName(USERNAME_TOKEN_QNAME_LPART),
+                        USERNAME_TOKEN + "-" + UUID.randomUUID().toString());
 
-    public boolean handleMessage(SOAPMessageContext smc) {
-        Boolean outbound = ((Boolean) smc.get(MessageContext.MESSAGE_OUTBOUND_PROPERTY));
-        SOAPMessage msg = smc.getMessage();
-        if (outbound) {
-            try {
-                ProcessingContext context = xwssProcessor.createProcessingContext(msg);
-                context.setSOAPMessage(msg);
-                SOAPMessage secureMsg = xwssProcessor.secureOutboundMessage(context);
-                smc.setMessage(secureMsg);
-            } catch (Exception e) {
-                throw new RuntimeException(e);
+                SOAPElement usernameElement = usernameToken.addChildElement(USERNAME, WSS_AUTH_PREFIX);
```

Comment:
Does SOAP encodes the username and password internally? if not please add te encoding logic for both username and password.

Reply:
As the server accepts plain text in the header, we cannot introduce any encoding, unfortunately!